cassiopeia/simulator/SimpleFitSubcloneSimulator.py

```diff
@@ -1,67 +1,91 @@
+    generates both deterministic and stochastic phylogenies and (2) the
+    phylogenies are as simple as possible while being non-neutral. The ability
+    to control the branch lengths of the neutrally evolving population and of
+    the fit subpopulation allows us to simulate different degrees of fitness
+    as well as stochasticity.
 
     Note that the branches for the leaves of the tree need not have length
-    exactly equal to branch_length_neutral nor branch_length_fit, because the
-    cells get assayed at the arbitrary time specified by 'experiment_duration',
-    which is not necessarily the time at which they divide.
+    exactly equal to 'branch_length_neutral' nor 'branch_length_fit', because
+    the population gets assayed at the arbitrary time specified by
+    'experiment_duration', which is not necessarily the time at which a
+    division event happens.
 
     The names of the leaf nodes will be given by a unique integer ID, followed
-    by "_neutral" for the neutrally evolving cells, and by "_fit" for the fit
-    cells.
+    by "_neutral" for the neutrally evolving individuals, and by "_fit" for the
+    fit individuals.
 
-    Finally, note that we do not enforce the condition branch_length_neutral >
-    branch_length_fit, therefore it is possible to make the subclone less fit,
-    i.e. expand slower.
+    Finally, note that it is possible to make the subclone less fit, i.e.
+    expand slower, for there is no constraint in the branch lengths, although
+    the most typical use case is the simulation of a more fit subpopulation.
 
     Args:
-        branch_length_neutral: Branch length of the neutrally evolving cells.
-            All cells are neutrally evolving until generation
+        branch_length_neutral: Branch length of the neutrally evolving
+            individuals. All individuals are neutrally evolving until generation
             'generations_until_fit_subclone', when exactly one of the lineages
-            develops fitness.
+            develops fitness. A generator can be provided instead of a constant,
+            which is useful for simulating random branch lengths.
         branch_length_fit: The branch length of the fit subclone, which appears
-            exactly at generation 'generations_until_fit_subclone'.
+            exactly at generation 'generations_until_fit_subclone'. A generator
+            can be provided instead of a constant, which is useful for
+            simulating random branch lengths.
         experiment_duration: The total length of the experiment.
         generations_until_fit_subclone: The generation at which one lineage
             develops fitness, giving rise to a fit subclone.
     """
 
     def __init__(
         self,
-        branch_length_neutral: float,
-        branch_length_fit: float,
+        branch_length_neutral: Union[float, Generator[float, None, None]],
+        branch_length_fit: Union[float, Generator[float, None, None]],
```

Comment:
I'm a bit confused why we have to use `Generators` here? The way I had imagined doing this is by passing some function for returning a branch lengths.

For example -- in the deterministic case: 

```
branch_length_neutral = lambda: 1
branch_length_neutral = lambda: 0.5
```

And in the case where you'd like to draw from some distribution:

```
branch_length_neutral = lambda: np.random.exponential(1)
branch_length_neutral = lambda: np.random.exponential(0.5)
```

Why is a `Generator` preferred to this? Is this because you'd like to support users just passing in a float rather than an anonymous function that always returns a constant?

Reply:
You're totally right Matt, we don't need to use a generator at all! We can use a callable. I'll make your proposed change, I agree it's much simpler.